hpx/runtime/applier/applier.hpp

```diff
@@ -159,7 +159,8 @@ namespace hpx { namespace applier
         }
 
         /// Schedule  threads based on the given parcel
-        void schedule_action(parcelset::parcel p, std::size_t num_thread = -1);
+        void schedule_action(parcelset::parcel p, std::size_t num_thread =
+                             (std::numeric_limits<std::size_t>::max)());
```

Comment:
@K-ballo WRT being idiomatic: see Thomas' first comment. Also, writing numeric_limits<foo>::max() hardly qualifies as "all the cruft".